tools/distpartitioning/utils.py

```diff
@@ -0,0 +1,260 @@
+    node_map = {}
+    x = metadata_list[0]["node_map"]
+    for k in x:
+        node_map[k] = []
+        for idx in range(len(metadata_list)):
+            node_map[k].append(metadata_list[idx]["node_map"][k][0])
+    graph_metadata["node_map"] = node_map
+
+    graph_metadata["ntypes"] = metadata_list[0]["ntypes"]
+    graph_metadata["num_edges"] = sum([metadata_list[i]["num_edges"] for i in range(len(metadata_list))])
+    graph_metadata["num_nodes"] = sum([metadata_list[i]["num_nodes"] for i in range(len(metadata_list))])
+    graph_metadata["num_parts"] = metadata_list[0]["num_parts"]
+    graph_metadata["part_method"] = metadata_list[0]["part_method"]
+
+    for i in range(len(metadata_list)):
+        graph_metadata["part-{}".format(i)] = metadata_list[i]["part-{}".format(i)]
+
+    with open('{}/{}.json'.format(output_dir, graph_name), 'w') as outfile: 
+        json.dump(graph_metadata, outfile, sort_keys=True, indent=4)
+
+def augment_edge_data(edge_data, part_ids):
+    """Add partition-id (rank which owns an edge) column to the edge_data.
+    
+    Parameters:
+    -----------
+    edge_data : numpy ndarray
+        Edge information as read from the xxx_edges.txt file
+    part_ids : numpy array
+        array of part_ids indexed by global_nid
+    Returns:
+    --------
+    numpy ndarray
+        original edge_data with one additional column
+    """
+    #tgtrank = [partitions[x[1]] for x in edge_data]
+    proc_ids = part_ids[edge_data[:,1]]
+    return np.c_[edge_data, proc_ids]
+
+def augment_node_data(node_data, part_ids): 
+    """Utility function to add auxilary columns to the node_data numpy ndarray.
+
+    Parameters:
+    -----------
+    node_data : numpy ndarray
+        Node information as read from xxx_nodes.txt file
+    part_ids : numpy array 
+        array of part_ids indexed by global_nid
+
+    Returns:
+    --------
+    numpy ndarray
+        original node_data, numpy ndarray, with two additional columns
+    """
+    global_nids = np.arange(node_data.shape[0], dtype=np.int64)
+    node_data_aug = np.c_[node_data, global_nids]
+
+    #proc_id = [partitions[x[6]] for x in node_data_aug]
+    proc_ids = part_ids[node_data_aug[:,6]]
+    return np.c_[node_data_aug, proc_ids]
+
+def read_nodes_file(nodes_file):
+    """Utility function to read xxx_nodes.txt file
+    
+    Parameters:
+    -----------
+    nodesfile : string
+        Graph file for nodes in the input graph
+    
+    Returns:
+    --------
+    numpy ndarray
+        node data as in the xxx_nodes.txt file
+    """
+    if nodes_file == "" or nodes_file == None:
+        return None
+
+    # Read the file from here.
+    # Assuming the nodes file is a numpy file
+    # nodes.txt file is of the following format
+    # <node_type> <weight1> <weight2> <weight3> <weight4> <global_type_nid> <attributes>
+    # For the ogb-mag dataset, nodes.txt is of the above format.
+    #
+    nodes_data = np.loadtxt(nodes_file, delimiter=' ', dtype='int64')
```

Comment:
What about attributes? Will they be float32?

Reply:
attributes are not used.

Reply:
attributes are not used.